### PR TITLE
Add project completion validation

### PR DESCRIPTION
**Pull request by lozette**
## Changes

A Project cannot be completed until All conditions have been met, the Grant payment certificate is received and the academy has opened. If the user tries to complete the project when these conditions are not met, they will see a message.

Screenshot 2023-06-01 at 15 45 14

## Checklist

- [x] Attach this pull request to the appropriate card in Trello.
- [x] Update the `CHANGELOG.md` if needed.

### CODE REVIEW

**Comment by lozette** [comment]
[Add project completion validation](https://trello.com/c/GXgeeOfv/1618-add-project-completion-validation)

File: config/locales/project.en.yml [hunk 1]
```diff
@@ -110,6 +110,7 @@ en:
           <p>You will not be able to change this project once it is completed.</p>
       submit_button: Complete project
       back_link: Return to project list
+      unable_to_complete: This project cannot be completed until all required conditions are met and the academy has opened.
```
Comment (mec) [inline config/locales/project.en.yml:113]: I think we should state all the criteria like you have in the commit message, as a list.

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ### Added
 
 - Display generated DfE number in project information view for the establishment
+- Do not allow a project to be completed until all conditions have been met and
```
Comment (mec) [inline CHANGELOG.md:12]: We should mention the grant support certificate has been returned as well.

File: app/models/conversion/project.rb [hunk 3]
```diff
@@ -19,4 +19,8 @@ def fetch_provisional_conversion_date
   def all_conditions_met?
     tasks_data.conditions_met_confirm_all_conditions_met?
   end
+
+  def opened?
+    !conversion_date_provisional? && conversion_date.past?
```
Comment (mec) [inline app/models/conversion/project.rb:24]: I know this is super pedantic! 

Can we change this to `conversion_date_confirmed_and_passed?` or something - I am nervous about this whole 'has the school opened' thing and I would rather it was not baked into the code until we get a straight answer.

**Review by mec — CHANGES_REQUESTED** [review]
Couple of small things -> Beautiful set of commits though - so clear! ❤️

**Review by mec — APPROVED** [review]
🚢

**Pull request merged**